tuplex/codegen/include/TypeAnnotatorVisitor.h

```diff
@@ -101,21 +102,25 @@ namespace tuplex {
                                        const python::Type& b);
         void assignHelper(NIdentifier *id, python::Type type);
         void checkRetType(python::Type t);
-        CompileError _typeError;
+        /*!
+         * Annotate iterator-related NCall with iterator-specific info
+         * @param funcName
+         * @param call
+         */
+        void annotateIteratorRelatedCalls(std::string &funcName, NCall* call);
```

Comment:
const std::string& funcName? If this is getting changed, then better use string as return value.

Reply:
Refactored.